backend/api.py

```
@@ -683,12 +715,110 @@ async def stream_answer_endpoint(request: SearchRequest, req: Request):
+
+# -------------------------------------------------------------------------
+# External Provider endpoints (Ollama, LM Studio, OpenAI-compatible)
+# -------------------------------------------------------------------------
+
+class ProviderQueryRequest(BaseModel):
+    provider_type: str  # 'ollama' or 'lmstudio'
+    base_url: Optional[str] = None
+    api_key: Optional[str] = ""
+
+@app.post("/api/providers/health")
+async def provider_health_check(body: ProviderQueryRequest, request: Request):
```

Ruff (0.15.10)

[warning] Unused function argument: `request`

(ARG001)